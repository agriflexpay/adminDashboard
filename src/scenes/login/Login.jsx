import axios from "axios";
import { Box, TextField, Button } from "@mui/material";
import { Formik } from "formik";
import * as yup from "yup";
import Header from "../../components/Header";
import {useAuth} from "../../AUTH/AuthContext"

const checkoutSchema = yup.object().shape({
  email: yup.string().email("invalid email").required("required"),
  password: yup.string().required("required"),
});
const LOGIN_URL = 'http://localhost:6002/api/user/login'

const initialValues = {
  email: "",
  password: "",
};


function Login() {
  const [showPassword, setShowPassword] = React.useState(false);
  const handleClickShowPassword = () => setShowPassword((show) => !show);
  const handleMouseDownPassword = () => setShowPassword(false);
  const {login}= useAuth()
  const handleFormSubmit = async (values, { setSubmitting })  => {
    const data = {
      email: values.email,
      password: values.password,
    };
   
    await axios.post(LOGIN_URL, data, {
      headers: {
        'Content-Type': 'application/json',
      },
      withCredentials: true,
    })
      .then((res) => {
     
          if (res.status === 200) {
            login(res.data.data)
            
          }
      })
      .catch((err) => {
        console.log(err);
        
      });


  };

  return (
    <Box
      display="flex"
      flexDirection="column"
      justifyContent="center"
      alignItems="center"
      height="100vh"
      mx="auto" // Center horizontally
    >
      <Header title="Login" subtitle="Enter your login details and submit" />

      <Formik
        onSubmit={handleFormSubmit}
        initialValues={initialValues}
        validationSchema={checkoutSchema}
      >
        {({
          values,
          errors,
          touched,
          handleBlur,
          handleChange,
          handleSubmit,
        }) => (
          <form onSubmit={handleSubmit}>
            <Box
              display="flex"
              flexDirection="column"
              justifyContent="center"
              alignItems="center"
            >
              <TextField
                fullWidth
                id="email"
                name="email"
                label="Email"
                type="email"
                variant="outlined"
                value={values.email}
                onChange={handleChange}
                onBlur={handleBlur}
                sx={{ mb: 2 }}
                error={touched.email && Boolean(errors.email)}
                helperText={touched.email && errors.email}
              />

              <TextField
                fullWidth
                id="password"
                name="password"
                label="Password"
                type="password"
                variant="outlined"
                value={values.password}
                onChange={handleChange}
                onBlur={handleBlur}
                sx={{ mb: 2, }}
                error={touched.password && Boolean(errors.password)}
                helperText={touched.password && errors.password}
              />
            </Box>

            <Box mt="20px">
              <Button type="submit" variant="contained" color="secondary" sx={{width:"20rem"}}>
                Submit
              </Button>
            </Box>
          </form>
        )}
      </Formik>
    </Box>
  );
}

export default Login;
