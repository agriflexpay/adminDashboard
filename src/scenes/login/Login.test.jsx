import React from 'react';
import { render, fireEvent, waitFor,screen, getByRole } from '@testing-library/react';
import renderer from 'react-test-renderer';
import axios from 'axios'; // Import axios for mocking
import '@testing-library/jest-dom'; 
import Login from './Login';
jest.mock('axios');
describe('Login', () => {
  //test cases here
  test('renders Login component', () => {
    render(<Login />);
  });

  test('renders the form', () => {
    const { getByLabelText, getByText } = render(<Login />);
    expect(getByLabelText('Email')).toBeInTheDocument();
    expect(getByLabelText('Password')).toBeInTheDocument();
    expect(getByText('Submit')).toBeInTheDocument();
  });


  test('Login component renders correctly', () => {
    const tree = renderer
      .create(<Login />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('fills out and submits the form', async () => {
    const { getByLabelText, getByText } = render(<Login />);

    // Fill out the form
    fireEvent.change(getByLabelText('Email'), { target: { value: 'test@example.com' } });
    fireEvent.change(getByLabelText('Password'), { target: { value: 'password123' } });

    // Mock the API call
    axios.post.mockResolvedValueOnce({ status: 200, data: { data: 'mock data' } });

    // Submit the form
    fireEvent.click(getByText('Submit'));

    // Wait for the async call to complete
    await waitFor(() => expect(axios.post).toHaveBeenCalled());

    // Expect login function to be called
    // You may need to mock the login function and test its invocation
    // For example:
    // expect(mockLogin).toHaveBeenCalledWith('mock data');
  });
  
  test('displays error message for invalid email', async () => {
    const { getByRole,getByLabelText} = render(<Login />);
  
    // Fill out the form with an invalid email
    fireEvent.change(getByLabelText('Email'), { target: { value: 'testmail@admin.com' } });
    fireEvent.change(getByLabelText('Password'), { target: { value: 'password123' } });
    // Mock the API call for invalid email
    axios.post.mockRejectedValueOnce({ response: { data: { message: 'Invalid email or password' } } });
  
    // Submit the form
    fireEvent.click(screen.getByText('Submit'));
  
    // Wait for the async call to complete
    await waitFor(() => expect(axios.post).toHaveBeenCalled());
  
    // Expect Email field to have aria-invalid attribute set to true
    expect(getByLabelText('Email')).toHaveAttribute('aria-invalid', 'false');
  });
});
