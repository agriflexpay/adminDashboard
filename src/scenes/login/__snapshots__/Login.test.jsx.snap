// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Login component renders correctly 1`] = `
[
  <div
    className="Toastify"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  />,
  <div
    className="MuiBox-root css-9g7dug"
  >
    <div
      className="MuiBox-root css-1xn3edc"
    >
      <h2
        className="MuiTypography-root MuiTypography-h2 css-be5vtk-MuiTypography-root"
      >
        Login
      </h2>
      <h5
        className="MuiTypography-root MuiTypography-h5 css-b1sixc-MuiTypography-root"
      >
        Enter your login details and submit
      </h5>
    </div>
    <form
      onSubmit={[Function]}
    >
      <div
        className="MuiBox-root css-wkpw2c"
      >
        <div
          className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-1j5gcdd-MuiFormControl-root-MuiTextField-root"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink={false}
            htmlFor="email"
            id="email-label"
          >
            Email
          </label>
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              disabled={false}
              id="email"
              name="email"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="email"
              value=""
            />
            <fieldset
              aria-hidden={true}
              className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="css-yjsfm1"
              >
                <span>
                  Email
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-1j5gcdd-MuiFormControl-root-MuiTextField-root"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink={false}
            htmlFor="password"
            id="password-label"
          >
            Password
          </label>
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              disabled={false}
              id="password"
              name="password"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="password"
              value=""
            />
            <fieldset
              aria-hidden={true}
              className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="css-yjsfm1"
              >
                <span>
                  Password
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        className="MuiBox-root css-1xaekgw"
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium css-1m2oesh-MuiButtonBase-root-MuiButton-root"
          disabled={false}
          onBlur={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="submit"
        >
          Submit
        </button>
      </div>
    </form>
  </div>,
]
`;
